Formula/g/gap.rb

```diff
@@ -0,0 +1,38 @@
+class Gap < Formula
+  desc "Groups, Algorithms, Programming - A system for computational discrete algebra"
+  homepage "https://www.gap-system.org"
+  url "https://github.com/gap-system/gap/releases/download/v4.13.0/gap-4.13.0.tar.gz"
+  sha256 "cc76ecbe33d6719450a593e613fb87e9e4247faa876f632dd0f97c398f92265d"
+  license "GPL-2.0-or-later"
+
+  depends_on "autoconf" => :build
+  depends_on "libtool" => :build
+  depends_on "gmp"
+  depends_on :macos
```

Comment:
Why does it specifically depend on macOS?

Reply:
GAP has a fairly different build for Mac vs Linux. I'm developing on Mac so only did the macOS option. I could try setting up separate linux / Mac dependencies and builds if that's necessary - I figured that could be added later by myself or someone else.

Reply:
Let's try and build on Linux too, since there is nothing technically blocking it.

Reply:
OK, I do have 2 Linux servers I can dev this on - I'll add Home brew to the Ubuntu box and get this working. Sorry for the delay busy lately.

Reply:
I have it working on my dev machine (Mac) and linux server. Not quite sure how ideally I've dealt with the dependencies: for Mac and Linux I added a 'gcc' and 'make' dependency ... the tool will however use any that it finds in the system path - that dependency will ensure I can run based on my understanding of home-brew gcc. However, this doesn't force the usage of hombrews gcc.